### DISCUSSION TITLE
Is the health check status available on an api endpoint?

### DISCUSSION THREAD

**Discussion by hansmbakker** [dotnet/yarp#1870] [Q&A] [answered]
There is health check functionality available in YARP. As I understood, it is used to decide what backend send traffic to.

Is this health check status also available [like in Asp.Net Core](https://learn.microsoft.com/en-us/aspnet/core/host-and-deploy/health-checks?view=aspnetcore-6.0)?

**Comment by Tratcher** [comment, marked as answer]
YARP doesn't currently make its own health status available externally. Since YARP plugs into an AspNetCore app, any health reporting would need to be combine with the rest of the app's health. See https://github.com/microsoft/reverse-proxy/issues/125. It should be possible to directly examine Destination's health state today, but you'd need to decide how to report that externally.
